components/cosmetic_filters/renderer/cosmetic_filters_js_handler.cc

```diff
@@ -77,88 +77,24 @@ const char kCosmeticFilteringInitScript[] =
 
 const char kHideSelectorsInjectScript[] =
     R"((function() {
-          let nextIndex =
-              window.content_cosmetic.cosmeticStyleSheet.rules.length;
           const selectors = %s;
           selectors.forEach(selector => {
             if ((typeof selector === 'string') &&
                 (window.content_cosmetic.hide1pContent ||
                 !window.content_cosmetic.allSelectorsToRules.has(selector))) {
               let rule = selector + '{display:none !important;}';
-              window.content_cosmetic.cosmeticStyleSheet.insertRule(
-                `${rule}`, nextIndex);
+              let ruleIndex = 0;
               if (!window.content_cosmetic.hide1pContent) {
+                ruleIndex = window.content_cosmetic.nextRuleIndex;
+                window.content_cosmetic.nextRuleIndex++;
                 window.content_cosmetic.allSelectorsToRules.set(
-                  selector, nextIndex);
+                  selector, ruleIndex);
                 window.content_cosmetic.firstRunQueue.add(selector);
               }
-              nextIndex++;
+              window.cf_worker.injectStylesheet(rule, ruleIndex);
             }
           });
-          if (!document.adoptedStyleSheets.includes(
-              window.content_cosmetic.cosmeticStyleSheet)) {
-            document.adoptedStyleSheets =
-              [window.content_cosmetic.cosmeticStyleSheet,
-                ...document.adoptedStyleSheets];
-          };
-        })();)";
-
-const char kForceHideSelectorsInjectScript[] =
-    R"((function() {
-          let nextIndex =
-              window.content_cosmetic.cosmeticStyleSheet.rules.length;
-          const selectors = %s;
-          selectors.forEach(selector => {
-            if (typeof selector === 'string') {
-              let rule = selector + '{display:none !important;}';
-              window.content_cosmetic.cosmeticStyleSheet.insertRule(
-                `${rule}`, nextIndex);
-              if (!window.content_cosmetic.hide1pContent) {
-                window.content_cosmetic.allSelectorsToRules.set(
-                  selector, nextIndex);
-              }
-              nextIndex++;
-            }
-          });
-          if (!document.adoptedStyleSheets.includes(
-              window.content_cosmetic.cosmeticStyleSheet)) {
-            document.adoptedStyleSheets =
-              [window.content_cosmetic.cosmeticStyleSheet,
-                ...document.adoptedStyleSheets];
-          };
-        })();)";
-
-const char kStyleSelectorsInjectScript[] =
```

Comment:
I've removed `kForceHideSelectorsInjectScript` and `kStyleSelectorsInjectScript` entirely. They were previously adding rules to `window.content_cosmetic.cosmeticStyleSheet` in order to keep track of the indices of rules, but this is no longer necessary as a result of using `injectStylesheet` with an id of `0`. At that point, there is no special behavior in JS and we can construct the corresponding stylesheets directly in C++ instead.

Reply:
cool refactor!